docs/source/upgrade_notes.rst

```diff
@@ -3,6 +3,13 @@
 Upgrade Notes
 =============
 
+.. _ref-upgrade-notes-v3-4dev:
+
+|st2| v3.4dev
```

Comment:
```suggestion
|st2| v3.4
```
As we're preparing the documentation for upcoming 3.4 release. 3.4dev is irrelevant to the users.